test/tools/router_check/validation.proto

```diff
@@ -0,0 +1,99 @@
+syntax = "proto3";
```

Comment:
Could this file benefit from any schema validators? https://github.com/envoyproxy/protoc-gen-validate. See other examples in the repo. If we use them (which we probably should) you will need to use some of the validation functions here: https://github.com/envoyproxy/envoy/blob/master/source/common/protobuf/utility.h#L221